noxfile.py

```diff
@@ -510,7 +515,7 @@ def docs(session):
 def docfx(session):
     """Build the docfx yaml files for this library."""
 
-    session.install("-e", ".")
+    session.install("-e", ".[scikit-learn]")
```

Comment:
I'm very concerned if we need to install `scikit-learn` just to build the docs. Please remove.

```suggestion
    session.install("-e", ".")
```

Reply:
Not sure if it's possible to remove this as we are calling publish_api_coverage.py, and one of the things it do is go through sklearn APIs and generate a coverage report.